oauth2.md

```diff
@@ -0,0 +1,205 @@
+# OAuth 2.0 Integration
+
+This document provides the steps for partners to integrate their applications to Mocaverse using OAuth 2.0 (Authorization Code Flow).
+
+## Steps
+
+### 1. Application Registration
+
+- Provide `redirect_uri` values to our team (e.g. https://your-app.example.com/callback)
+- Obtain the `client_id` and `client_secret` from our team
+
+### 2. User Authorization Request
+
+When performing authorization request, specify the following parameters
+- `response_type`: must be set to `code`
+- `client_id`: client_id from step 1
+- `redirect_uri`: One of the registered `redirect_uri` from step 1
+- `state`: a random string generated by your app to prevent CSRF attacks
+
+```html
+<!--
+Example Values:
+- response_type: code
+- client_id: b6084afa-e9ee-4eb4-8220-338be0c12374
+- redirect_uri: https://your-app.example.com/callback
+- state: v6glrJn3gf3qL4rPFLBB
+-->
+
+<a href="https://oauth.mocaverse.xyz/api/oauth/auth?response_type=code&client_id=b6084afa-e9ee-4eb4-8220-338be0c12374&redirect_uri=https%3A%2F%2Fyour-app.example.com%2Fcallback&state=v6glrJn3gf3qL4rPFLBB">
+  Login with Mocaverse
+</a>
+```
+
+#### Response
+
+| Response | Description                                                                     | Sample Return             |
```

Comment:
`Sample Response` instead of `Sample Return` - same as all below

add a table describing each response field